src/test/java/utils/TestResultsWatcher.java

```diff
@@ -0,0 +1,45 @@
+package utils;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.TestWatcher;
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+import java.util.Optional;
+
+public class TestResultsWatcher implements TestWatcher {
+    private final WebDriver driver;
+    private final String fileName;
+    private final Capabilities capabilities;
+
+    public TestResultsWatcher() {
+        this.fileName = RandomStringUtils.randomAlphabetic(10);
+        this.driver = WebDriverSingleton.getDriver();
+        this.capabilities = ((RemoteWebDriver) driver).getCapabilities();
```

Comment:
Driver should be removed from constructor. You will have issues with listener class, cause it possible that you will use instance of browser which was closed.
You don't need to save it in class variables. Just call WebDriverSingleton.getDriver() directly in the method where you need it.

Same is applicable for RandomStringUtils.randomAlphabetic(10)

Reply:
resolved